pulpcore/pulpcore/app/serializers/task.py

```diff
@@ -115,10 +115,17 @@ class WorkerSerializer(ModelSerializer):
     )
 
     online = serializers.BooleanField(
-        help_text='Whether the worker is online or not. Defaults to True.',
+        help_text=_('Whether the worker is online'),
+        read_only=True
+    )
+
+    gracefully_stopped = serializers.BooleanField(
+        help_text=_('If this worker is currently offline, this represents whether the worker \
+            was shut down properly, as opposed to e.g. crashed'),
         read_only=True
     )
 
     class Meta:
         model = models.Worker
-        fields = ModelSerializer.Meta.fields + ('name', 'last_heartbeat', 'online')
+        fields = ModelSerializer.Meta.fields + ('name', 'last_heartbeat',
```

Comment:
What does this line do? It looks right but I want to learn more.

Reply:
Here's the docs page: http://www.django-rest-framework.org/api-guide/serializers/#specifying-which-fields-to-include

It looks like it's ensuring that all of the fields specified to include on the base ModelSerializer are included on the final Model, but I'm not sure if there is a better way to accomplish this.  Maybe worth looking into.

Reply:
Thank you!